commands/deploy.go

```diff
@@ -212,7 +215,11 @@ Error: %s`, fprocessErr.Error())
 				Requests: function.Requests,
 			}
 
-			proxy.DeployFunction(function.FProcess, services.Provider.GatewayURL, function.Name, function.Image, function.RegistryAuth, function.Language, deployFlags.replace, allEnvironment, services.Provider.Network, functionConstraints, deployFlags.update, deployFlags.secrets, allLabels, functionResourceRequest1)
+			statusCode := proxy.DeployFunction(function.FProcess, services.Provider.GatewayURL, function.Name, function.Image, function.RegistryAuth, function.Language, deployFlags.replace, allEnvironment, services.Provider.Network, functionConstraints, deployFlags.update, deployFlags.secrets, allLabels, functionResourceRequest1)
+			functionsStatusCodes[k] = statusCode
+			if err := deployFailed(functionsStatusCodes); err != nil {
```

Comment:
I'm not sure what this code is doing.

You should check the status code here and then if there is an error collect it, so at the end of the code we have a list of errors that can be printed to the client or returned as a single error.